app/reducers/surveys.ts

```diff
@@ -70,24 +66,76 @@ export const selectSurveyTemplates = createSelector(
 );
 
 type SurveyTemplateProps = {
-  templateType: NonNullable<UnknownSurvey['templateType']>;
+  templateType: EventType;
+};
+export type SurveyTemplate = Omit<
+  DetailedSurvey,
+  'id' | 'event' | 'activeForm' | 'templateType'
+> & {
+  questions: (Omit<DetailedSurvey['questions'][number], 'id'> & {
+    options: Omit<DetailedSurvey['questions'][number]['options'], 'id'>;
+  })[];
+  templateType: EventType;
 };
 
 export const selectSurveyTemplate = createSelector(
   (state: RootState) => selectSurveys(state),
-  (state, props: SurveyTemplateProps) => props.templateType,
+  (_: RootState, props: SurveyTemplateProps) => props.templateType,
   (surveys, templateType) => {
     const template = surveys.find(
       (survey) => survey.templateType === templateType
     );
-    if (!template) return false;
+    if (!template) return undefined;
     const questions = (template.questions || []).map((question) => ({
       ...omit(question, 'id'),
       options: question.options.map((option) => omit(option, 'id')),
     }));
     return {
-      ...omit(template, ['id', 'event', 'activeFrom', 'templateType']),
+      ...omit(template, ['id', 'event', 'activeFrom']),
       questions,
-    };
+    } as SurveyTemplate;
   }
 );
+
+export const useFetchedTemplate = (
+  prepareId: string,
+  templateType?: EventType
+): SurveyTemplate | undefined => {
+  const dispatch = useAppDispatch();
+  usePreparedEffect(
+    `useFetchedTemplate-${prepareId}`,
+    () => templateType && dispatch(fetchTemplate(templateType)),
+    [templateType]
+  );
+  return useAppSelector((state: RootState) =>
+    templateType ? selectSurveyTemplate(state, { templateType }) : undefined
+  );
+};
+
+export function useFetchedSurvey(
+  prepareId: string,
+  surveyId: ID,
+  token: string
+): SelectedPublicResultsSurvey | undefined;
+export function useFetchedSurvey(
+  prepareId: string,
+  surveyId: ID
+): SelectedSurvey | undefined;
+export function useFetchedSurvey(
+  prepareId: string,
+  surveyId: ID,
+  token?: string
+) {
```

Comment:
Overloading the function to get correct type for the public survey results page.
If you provide a token, it will fetch the public results which have a completely different type from the normal authenticated-page surveys.

Reply:
Huh, that's quite cool. I didn't know you could do that